scripts/zkai_attention_derived_d128_projection_boundary_gate.py

```
@@ -0,0 +1,940 @@
+from __future__ import annotations
+
+import argparse
+import copy
+import csv
+import hashlib
+import importlib.util
+import io
+import json
+import pathlib
+import sys
+from typing import Any, Callable
```

Ruff (0.15.12)

[warning] Import from `collections.abc` instead: `Callable`

Import from `collections.abc`

(UP035)